### PR TITLE
I've created the modules for de api recurring invoices controller and…

### PR DESCRIPTION
**Pull request by RodrigoMolinaMadrid**
… for the recurring invoices context.

The mix dialyzer return some fails beacuse the reccurring invoice schema is not created yet.
I've also added the new paths to the router file

### CODE REVIEW

File: lib/siwapp_web/controllers/api/recurring_invoices_controller.ex [hunk 1]
```diff
@@ -0,0 +1,40 @@
+defmodule SiwappWeb.RecurringInvoicesController do
```
Comment (peillis) [inline lib/siwapp_web/controllers/api/recurring_invoices_controller.ex:1]: You should name these API modules like `SiwappWeb.Api.RecurringInvoicesController` and the same with the InvoicesController. This way we'll know they are API controllers and not HTML controllers.

File: lib/siwapp_web/router.ex [hunk 2]
```diff
@@ -37,12 +37,16 @@ defmodule SiwappWeb.Router do
...
     put "/invoices/:id", InvoicesController, :update
-
     delete "/invoices/:id", InvoicesController, :delete
+
+    get "/recurring_invoices", RecurringInvoicesController, :list
+    get "/recurring_invoices/show/:id", RecurringInvoicesController, :show
+    get "/recurring_invoices/:id", RecurringInvoicesController, :generate_invoices
+    post "/recurring_invoices", RecurringInvoicesController, :create
+    put "/recurring_invoices/:id", RecurringInvoicesController, :update
+    delete "/recurring_invoices/:id", RecurringInvoicesController, :delete
```
Comment (peillis) [inline lib/siwapp_web/router.ex:49]: Instead of all these routes you can use `resources`. See https://hexdocs.pm/phoenix/Phoenix.Router.html#resources/4

**Review by peillis — CHANGES_REQUESTED** [review]

**Comment by peillis** [comment]
By the way, the title for the PR is just for a brief title, not for explaining in detail what are you doing :)

File: lib/siwapp_web/controllers/api/recurring_invoices_controller.ex [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      {:ok, _} -> json(conn, "The invoice was successfully updated")
+      {:error, changeset} -> json(conn, changeset)
+    end
+  end
+
+  @spec delete(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def delete(conn, %{"id" => id}) do
+    recurring_invoice = RecurringInvoices.get!(id)
+    {:ok, response} = RecurringInvoices.delete(recurring_invoice)
+    json(conn, response)
```
Comment (olirods) [inline lib/siwapp_web/controllers/api/recurring_invoices_controller.ex:33]: Oh you finally did it with this function, it's great!

**Review by olirods — APPROVED** [review]

**Pull request closed without merging**